tests/ImageSharp.Tests/Formats/Jpg/JpegDecoderTests.cs

```diff
@@ -127,60 +128,52 @@ public async Task DecodeAsync_DegenerateMemoryRequest_ShouldTranslateTo_ImageFor
         }
 
         [Theory]
-        [InlineData(TestImages.Jpeg.Baseline.Jpeg420Small, 0)]
-        [InlineData(TestImages.Jpeg.Issues.ExifGetString750Transform, 1)]
-        [InlineData(TestImages.Jpeg.Issues.ExifGetString750Transform, 15)]
-        [InlineData(TestImages.Jpeg.Issues.ExifGetString750Transform, 30)]
-        [InlineData(TestImages.Jpeg.Issues.BadRstProgressive518, 1)]
-        [InlineData(TestImages.Jpeg.Issues.BadRstProgressive518, 15)]
-        [InlineData(TestImages.Jpeg.Issues.BadRstProgressive518, 30)]
-        public async Task Decode_IsCancellable(string fileName, int cancellationDelayMs)
+        [InlineData(0)]
+        [InlineData(0.5)]
+        [InlineData(0.9)]
+        public async Task Decode_IsCancellable(int percentageOfStreamReadToCancel)
         {
-            // Decoding these huge files took 300ms on i7-8650U in 2020. 30ms should be safe for cancellation delay.
-            string hugeFile = Path.Combine(
-                TestEnvironment.InputImagesDirectoryFullPath,
-                fileName);
-
-            const int numberOfRuns = 5;
-
-            for (int i = 0; i < numberOfRuns; i++)
+            var cts = new CancellationTokenSource();
+            var file = Path.Combine(TestEnvironment.InputImagesDirectoryFullPath, TestImages.Jpeg.Baseline.Jpeg420Small);
+            using var pausedStream = new PausedStream(file);
+            pausedStream.OnWaiting(s =>
             {
-                var cts = new CancellationTokenSource();
-                if (cancellationDelayMs == 0)
+                if (s.Position >= s.Length * percentageOfStreamReadToCancel)
                 {
                     cts.Cancel();
+                    pausedStream.Release();
                 }
                 else
                 {
-                    cts.CancelAfter(cancellationDelayMs);
+                    // allows this/next wait to unblock
+                    pausedStream.Next();
                 }
+            });
 
-                try
-                {
-                    using Image image = await Image.LoadAsync(hugeFile, cts.Token);
-                }
-                catch (TaskCanceledException)
-                {
-                    // Successfully observed a cancellation
-                    return;
-                }
-            }
+            AsyncLocalSwitchableFilesystem.ConfigureFileSystemStream(pausedStream);
```

Comment:
Since xunit tests are running in parallel, we should not modify any global state, but rather pass a new `Configuration` instance to `Image.LoadAsync`. It can be created by `Configuration.Default.DeepClone()`.

Reply:
That means we don't actually test our usage of the global context here.... This is exactly why I created this specific version of `IFileStream` that proxies to the same time as default global version in all cases unless in an async context where its specifically been overridden... this should be perfectly safe along as this is the only mechanism used to switch out the global file system.

Reply:
hmm...actually changed my mind, will switch over to the overload that take a configuration.